src/py/mat3ra/made/tools/modify.py

```diff
@@ -462,3 +467,202 @@ def rotate_material(material: Material, axis: List[int], angle: float) -> Materi
     atoms.wrap()
 
     return Material(from_ase(atoms))
+
+
+def get_passivant_coordinate_crystal(
+    material: Material, atom_coordinate: np.ndarray, bond_direction: np.ndarray, bond_length: float = 2.0
+):
+    bond_normal_cartesian = np.array(material.basis.cell.convert_point_to_cartesian(bond_direction.tolist()))
+    bond_normal_cartesian /= np.linalg.norm(bond_normal_cartesian)
+    bond_vector_cartesian = bond_normal_cartesian * bond_length
+    bond_vector_crystal = np.array(material.basis.cell.convert_point_to_crystal(bond_vector_cartesian.tolist()))
+    passivant_coordinate_crystal = atom_coordinate + bond_vector_crystal
+    return passivant_coordinate_crystal
+
+
+def passivate_material(
```

Comment:
consider simplifying to avoid duplicating the 3x3x3 supercell logic
@on_before(translate_to_supercell([3,3,1]))
@on_after(translate_to_supercell([3,3,1], reverse=True))